tests/transform/test_genes_biodomains.py

```diff
@@ -40,57 +46,51 @@ def test_count_grouped_total_two_groups(self):
-
-#     result_df = transform.standardize_column_names(df)
-#     assert type(result_df) is pd.core.frame.DataFrame
-#     assert list(result_df.columns) == ['team_id', 'team_name', 'team-sco']
-
-
-# def test_standardize_values():
-
-#     assert df.isna().sum().sum() == 2
-
-#     result_df = transform.standardize_values(df)
-
-#     assert type(result_df) is pd.core.frame.DataFrame
-#     assert result_df.isna().sum().sum() == 0
-#     assert result_df.shape == (4, 3)
-
-# def test_rename_columns():
-#     refresh_df = pd.DataFrame(
-#         {'team id': [np.nan, 0, 1, 2],
-#          'team.Name': ['MSN', 'Team 1', 'Team 2', np.nan],
-#          'team-Sco@#&': ['x', 'y', 'z', "na"]})
-
-#     bad_result_df = transform.rename_columns(df=refresh_df, column_map={"team-Sco@#&"})
-#     assert type(bad_result_df) is pd.core.frame.DataFrame
-#     assert list(bad_result_df.columns) == ["team id", "team.Name", "team-Sco@#&"]
-
-#     partial_good_result_df = transform.rename_columns(df=refresh_df, column_map={"team-Sco@#&": "team_scope"})
-#     assert list(partial_good_result_df.columns) == ['team id', 'team.Name', 'team_scope']
-#     assert type(partial_good_result_df) is pd.core.frame.DataFrame
+class TestTransformBiodomains:
+    data_files_path = "tests/test_assets/data_files"
+    test_data = [
+        (
+            "biodomains_test_input.csv",
+            "genes_biodomains.json",
+            True,
+        ),  # pass with good data
+        (  # pass with imperfect, but handled data
+            "biodomains_test_input_bad_but_should_pass.csv",
+            "genes_biodomains_bad_output_but_should_pass.json",
+            True,
+        ),
+        (
+            "biodomains_test_input_bad_should_fail.csv",
+            None,
+            False,
+        ),  # fail with bad data
+    ]
+
+    @pytest.mark.parametrize("input_file, expected_output_file, expect_pass", test_data)
+    def test_transform_biodomains(self, input_file, expected_output_file, expect_pass):
+        if expect_pass:
+            input_df = pd.read_csv(
+                os.path.join(self.data_files_path, "input", input_file)
+            )
+            output_df = genes_biodomains.transform_genes_biodomains(
+                datasets={"genes_biodomains": input_df}
+            )
+            expected_df = pd.read_json(
+                os.path.join(self.data_files_path, "output", expected_output_file),
+            )
+            pd.testing.assert_frame_equal(output_df, expected_df)
+            message = f"Execution on {input_file} was successful and matches the expected output."
+            logger.info(message)
+        else:
+            try:
+                input_df = pd.read_csv(
+                    os.path.join(self.data_files_path, "input", input_file)
+                )
+                genes_biodomains.transform_genes_biodomains(
+                    datasets={"genes_biodomains": input_df}
+                )
+            except KeyError as e:
```

Comment:
I recommend asserting that a key error is raised, this way when it isn't raised, it will fail the test.